src/Analyzers/CSharp/Analyzers/PreferTrailingComma/PreferTrailingCommaDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,87 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using Microsoft.CodeAnalysis.CodeStyle;
+using Microsoft.CodeAnalysis.CSharp.CodeStyle;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.PreferTrailingComma
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    internal sealed class PreferTrailingCommaDiagnosticAnalyzer : AbstractBuiltInCodeStyleDiagnosticAnalyzer
+    {
+        public PreferTrailingCommaDiagnosticAnalyzer() : base(
+            diagnosticId: IDEDiagnosticIds.PreferTrailingCommaDiagnosticId,
+            enforceOnBuild: EnforceOnBuildValues.PreferTrailingComma,
+            option: CSharpCodeStyleOptions.PreferTrailingComma,
+            language: LanguageNames.CSharp,
+            title: new LocalizableResourceString(nameof(CSharpAnalyzersResources.Missing_trailing_comma), CSharpAnalyzersResources.ResourceManager, typeof(CSharpAnalyzersResources)))
+        {
+        }
+
+        public override DiagnosticAnalyzerCategory GetAnalyzerCategory() => DiagnosticAnalyzerCategory.SyntaxTreeWithoutSemanticsAnalysis;
+
+        protected override void InitializeWorker(AnalysisContext context)
+        {
+            // TODO:
+            // list patterns
+            // anonymous object creation
+            // initializer expression
+            // switch expression
```

Comment:
📝 Marked as draft pending update to address test gaps

Reply:
Added support for all of these.